src/Engine/Spells/CastSpellInfo.cpp

```diff
@@ -350,11 +346,10 @@ void CastSpellInfoHelpers::castSpell() {
                             assert(false);
                     }
                     int spikes_active = 0;
-                    for (uint i = 0; i < pSpriteObjects.size(); ++i) {
-                        SpriteObject *object = &pSpriteObjects[i];
-                        if (object->uType &&
-                                object->uSpellID == SPELL_FIRE_FIRE_SPIKE &&
-                                object->spell_caster_pid == PID(OBJECT_Player, pCastSpell->uPlayerID)) {
+                    for (SpriteObject &ObjectRef : pSpriteObjects) {
```

Comment:
Codestyle: variables start with lowercase. Just `spriteObject` should do. Also you actually need `const SpriteObject &` here.